lib/ASLScalar.ml

```diff
@@ -275,11 +275,23 @@ let convert_to_bool = function
   | S_BitVector _ as s ->
       Warn.fatal "ASLScalar invalid op: to_bool %s" (pp false s)
 
-let convert_to_bv = function
+(** Convert to bitvector of known size *)
+let convert_to_bv sz = function
+  | S_BitVector bv as v ->
+      let len = BV.length bv in
+      if sz < len then S_BitVector (BV.prefix bv sz)
+      else if sz > len then
+        let zs = BV.zeros (sz-len) in
+        S_BitVector (BV.concat [bv;zs;])
```

Comment:
I'm not 100% sure here, but the order of the list elements passed `BV.concat` here seems to be the reverse to e.g. [this line](https://github.com/herd/herdtools7/blob/6f0695d73cc57d905f623cbcb4b892bf1616d847/lib/ASLScalar.ml#L247) which I think attempts a similar zero-extension

Reply:
You are right, thanks. Fixed.